server/src/api/feature/router.ts

```diff
@@ -24,30 +25,69 @@ export class FeatureRouter {
   }
 
   private async createFeature(req: Request, res: Response) {
-    const featureId = await this.controller.createFeature(req.body)
-    res.json({ status: 'ok', featureId: featureId })
+    try {
+      let token = req.headers['auth'] as string
+      const featureId = await this.controller.createFeature(req.body, token)
+      res.json({ status: 'ok', featureId: featureId })
+    } catch (e) {
+      if (e instanceof LoginError) {
+        res.status(403).json({ error: 'incorrect credentials' })
+        return
+      }
+    }
   }
 
-  private async getFeature(req: Request, res: Response) {
-    const id = parseInt(req.params.id)
-    if (id === null) {
-      res.status(400).json({ error: 'id is needed for feature call' })
+  public async getFeature(req: Request, res: Response) {
+    try {
+      let token = req.headers['auth'] as string
+      const id = parseInt(req.params.id)
+      if (id === null) {
+        res.status(400).json({ error: 'id is needed for feature call' })
+      }
+      res.json(await this.controller.getFeature(id, token))
+    } catch (e) {
+      if (e instanceof LoginError) {
+        res.status(403).json({ error: 'incorrect credentials' })
+        return
+      }
     }
-    res.json(await this.controller.getFeature(id))
   }
 
   private async updateFeature(req: Request, res: Response) {
-    await this.controller.updateFeature(req.body)
-    res.json({ status: 'ok' })
+    try {
+      let token = req.headers['auth'] as string
+      await this.controller.updateFeature(req.body, token)
+      res.json({ status: 'ok' })
```

Comment:
Nit, it's good to have consistent returns in functions, e.g. one code path returns while the other doesn't.